pallets/parachain-staking/src/lib.rs

```diff
@@ -0,0 +1,1870 @@
+            // insert empty top delegations
+            <TopDelegations<T>>::insert(&acc, empty_delegations.clone());
+            // insert empty bottom delegations
+            <BottomDelegations<T>>::insert(&acc, empty_delegations);
+            <CandidatePool<T>>::put(candidates);
+            let new_total = <Total<T>>::get().saturating_add(bond);
+            <Total<T>>::put(new_total);
+            Self::deposit_event(Event::JoinedCollatorCandidates {
+                account: acc,
+                amount_locked: bond,
+                new_total_amt_locked: new_total,
+            });
+            Ok(().into())
+        }
+        #[pallet::weight(<T as Config>::WeightInfo::schedule_leave_candidates(*candidate_count))]
+        /// Request to leave the set of candidates. If successful, the account is immediately
+        /// removed from the candidate pool to prevent selection as a collator.
+        pub fn schedule_leave_candidates(
+            origin: OriginFor<T>,
+            candidate_count: u32,
+        ) -> DispatchResultWithPostInfo {
+            let collator = ensure_signed(origin)?;
+            let mut state = <CandidateInfo<T>>::get(&collator).ok_or(Error::<T>::CandidateDNE)?;
+            let (now, when) = state.schedule_leave::<T>()?;
+            let mut candidates = <CandidatePool<T>>::get();
+            ensure!(
+                candidate_count >= candidates.0.len() as u32,
+                Error::<T>::TooLowCandidateCountToLeaveCandidates
+            );
+            if candidates.remove(&Bond::from_owner(collator.clone())) {
+                <CandidatePool<T>>::put(candidates);
+            }
+            <CandidateInfo<T>>::insert(&collator, state);
+            Self::deposit_event(Event::CandidateScheduledExit {
+                exit_allowed_round: now,
+                candidate: collator,
+                scheduled_exit: when,
+            });
+            Ok(().into())
+        }
+
+        #[pallet::weight(
+			<T as Config>::WeightInfo::execute_leave_candidates(*candidate_delegation_count)
+		)]
+        /// Execute leave candidates request
+        pub fn execute_leave_candidates(
+            origin: OriginFor<T>,
+            candidate: T::AccountId,
+            candidate_delegation_count: u32,
+        ) -> DispatchResultWithPostInfo {
+            ensure_signed(origin)?;
+            let state = <CandidateInfo<T>>::get(&candidate).ok_or(Error::<T>::CandidateDNE)?;
+            ensure!(
+                state.delegation_count <= candidate_delegation_count,
+                Error::<T>::TooLowCandidateDelegationCountToLeaveCandidates
+            );
+            state.can_leave::<T>()?;
+            let return_stake = |bond: Bond<T::AccountId, BalanceOf<T>>| -> DispatchResult {
+                // remove delegation from delegator state
+                let mut delegator = DelegatorState::<T>::get(&bond.owner).expect(
```

Comment:
expect, these OptionQuery should not expect or unwrap.

Reply:
what's the alternative, you have to get the value from the optionquery here

Reply:
return error